lib/minitest/queue.rb

```diff
@@ -4,6 +4,46 @@
 require 'minitest/reporters'
 
 module Minitest
+  class Requeue < Skip
+    attr_reader :failure
+
+    def initialize(failure)
+      super()
+      @failure = failure
+    end
+
+    def result_label
+      "Requeued"
+    end
+
+    def backtrace
+      failure.backtrace
+    end
+
+    def error
+      failure.error
+    end
+
+    def message
+      failure.message
+    end
+  end
+
+  module Requeueing
+    # Make requeues acts as skips for reporters not aware of the difference.
+    def skipped?
+      super || requeued?
+    end
```

Comment:
Is this method necessary? If a test is `requeued?` it should always be `skipped?` as well, right?